asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py

```diff
@@ -989,3 +989,76 @@ def download_cdd_data(protocol_name: str) -> pd.DataFrame:
     )
 
     return formatted_data
+
+
+def clean_result_network(network, console=None):
+    """
+    Cleans an incoming result network JSON file from some issues that might occur. Current procedures:
+    - removes edges that have DG==0.0, this happens when there is inconsistent stereo annotation in input ligands
+    such that after stereo enumeration there are duplicate ligands.
+    - cleans imbalanced complex/solvent legs, e.g. when some have failed or when stereo expansion was done unintionally
+
+    returns the loaded FreeEnergyCalculationNetwork.
+    """
+    import json
+    import sys
+    import tempfile
+
+    import numpy as np
+    import rich
+    from asapdiscovery.alchemy.schema.fec import FreeEnergyCalculationNetwork
+    from rich.padding import Padding
+
+    with open(network) as f:
+        d = json.load(f)
+        input_results = d["results"]["results"]
+
+    # 1. remove edges where DG is 0.0
+    cleaned_results = []
+    for result in input_results:
+        if not result["estimate"]["magnitude"] == 0.0:
+            cleaned_results.append(result)
```

Comment:
use list comp to make more compact

```suggestion
cleaned_results = [result for result in input_results if not result["estimate"]["magnitude"] == 0.0]
```